harbor-client/src/cashu_client.rs

```diff
@@ -0,0 +1,334 @@
+            .join_paths(&["v1", "mint", "quote", "bolt11"])?;
+        self.http_post(url, &request).await
+    }
+
+    /// Mint Quote status
+    async fn get_mint_quote_status(
+        &self,
+        quote_id: &str,
+    ) -> Result<MintQuoteBolt11Response<String>, Error> {
+        let url = self
+            .mint_url
+            .join_paths(&["v1", "mint", "quote", "bolt11", quote_id])?;
+
+        self.http_get(url).await
+    }
+
+    /// Mint Tokens [NUT-04]
+    async fn post_mint(
+        &self,
+        request: MintBolt11Request<String>,
+    ) -> Result<MintBolt11Response, Error> {
+        let url = self.mint_url.join_paths(&["v1", "mint", "bolt11"])?;
+        self.http_post(url, &request).await
+    }
+
+    /// Melt Quote [NUT-05]
+    async fn post_melt_quote(
+        &self,
+        request: MeltQuoteBolt11Request,
+    ) -> Result<MeltQuoteBolt11Response<String>, Error> {
+        let url = self
+            .mint_url
+            .join_paths(&["v1", "melt", "quote", "bolt11"])?;
+        self.http_post(url, &request).await
+    }
+
+    /// Melt Quote Status
+    async fn get_melt_quote_status(
+        &self,
+        quote_id: &str,
+    ) -> Result<MeltQuoteBolt11Response<String>, Error> {
+        let url = self
+            .mint_url
+            .join_paths(&["v1", "melt", "quote", "bolt11", quote_id])?;
+
+        self.http_get(url).await
+    }
+
+    /// Melt [NUT-05]
+    /// [Nut-08] Lightning fee return if outputs defined
+    async fn post_melt(
+        &self,
+        request: MeltBolt11Request<String>,
+    ) -> Result<MeltQuoteBolt11Response<String>, Error> {
+        let url = self.mint_url.join_paths(&["v1", "melt", "bolt11"])?;
+        self.http_post(url, &request).await
+    }
+
+    /// Swap Token [NUT-03]
+    async fn post_swap(&self, swap_request: SwapRequest) -> Result<SwapResponse, Error> {
+        let url = self.mint_url.join_paths(&["v1", "swap"])?;
+        self.http_post(url, &swap_request).await
+    }
+
+    /// Get Mint Info [NUT-06]
+    async fn get_mint_info(&self) -> Result<MintInfo, Error> {
+        let url = self.mint_url.join_paths(&["v1", "info"])?;
+        self.http_get(url).await
+    }
+
+    /// Spendable check [NUT-07]
+    async fn post_check_state(
+        &self,
+        request: CheckStateRequest,
+    ) -> Result<CheckStateResponse, Error> {
+        let url = self.mint_url.join_paths(&["v1", "checkstate"])?;
+        self.http_post(url, &request).await
+    }
+
+    /// Restore request [NUT-13]
+    async fn post_restore(&self, request: RestoreRequest) -> Result<RestoreResponse, Error> {
+        let url = self.mint_url.join_paths(&["v1", "restore"])?;
+        self.http_post(url, &request).await
+    }
+}
+
+pub fn spawn_lightning_payment_thread(
+    mut sender: Sender<CoreUIMsgPacket>,
+    client: Wallet,
+    storage: Arc<dyn DBConnection + Send + Sync>,
+    quote: MeltQuote,
+    msg_id: Uuid,
+    is_transfer: bool,
+) {
+    spawn(async move {
```

Comment:
idk if this is how we do it in the rest of the codebase, but is it possible for us to kill the whole app if an expect fails in a thread like this, rather than failing silently? only reason I knew what was going on with my last 2 panics was because I had logs open

Reply:
I am not too sure, it needs to be in a separate thread at the very least